src/Adyen/Service/NotificationReceiver.php

```diff
@@ -0,0 +1,229 @@
+<?php
+/**
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen API Library for PHP
+ *
+ * Copyright (c) 2020 Adyen N.V.
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ *
+ */
+
+namespace Adyen\Service;
+
+use Adyen\AdyenException;
+use Adyen\Exception\AuthenticationException;
+use Adyen\Exception\HMACKeyValidationException;
+use Adyen\Exception\MerchantAccountCodeException;
+use Adyen\Util\HmacSignature;
+use Psr\Log\LoggerInterface;
+
+class NotificationReceiver
+{
+
+    /**
+     * @var string
+     */
+    private $notificationHMAC;
+
+    /**
+     * @var HmacSignature
+     */
+    private $hmacSignature;
+
+    /**
+     * @var string
+     */
+    private $merchantAccount;
+
+    /**
+     * @var string
+     */
+    private $notificationUsername;
+
+    /**
+     * @var string
+     */
+    private $notificationPassword;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var Configuration
+     */
+    private $configuration;
+
+    /**
+     * @var AdyenNotification
+     */
+    private $adyenNotification;
+
+    /**
+     * NotificationReceiver constructor.
+     *
+     * @param HmacSignature $hmacSignature
+     * @param $notificationHMAC
+     * @param $merchantAccount
+     * @param $notificationUsername
+     * @param $notificationPassword
+     * @param LoggerInterface $logger
+     * @param Configuration $configuration
+     * @param AdyenNotification $adyenNotification
+     */
+    public function __construct(
+        HmacSignature $hmacSignature,
+        $notificationHMAC,
+        $merchantAccount,
+        $notificationUsername,
+        $notificationPassword,
+        LoggerInterface $logger, //TODO outsource?
+        Configuration $configuration, //TODO outsource?
```

Comment:
I would remove this and adjust the functions to receive the configuration in parameters